sklearnex/tests/_utils.py

```diff
@@ -0,0 +1,148 @@
+# ==============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from inspect import isclass
+
+import numpy as np
+from sklearn.base import (
+    BaseEstimator,
+    ClassifierMixin,
+    ClusterMixin,
+    OutlierMixin,
+    RegressorMixin,
+    TransformerMixin,
+)
+from sklearn.datasets import load_diabetes, load_iris
+from sklearn.neighbors._base import KNeighborsMixin
+
+from onedal.tests.utils._dataframes_support import _convert_to_dataframe
+from sklearnex import get_patch_map, patch_sklearn, unpatch_sklearn
+from sklearnex.neighbors import (
+    KNeighborsClassifier,
+    KNeighborsRegressor,
+    LocalOutlierFactor,
+    NearestNeighbors,
+)
+from sklearnex.svm import SVC, NuSVC
+
+
+def _load_all_models(patched, estimator=True):
+
+    try:
+        if patched:
+            patch_sklearn()
+
+        models = {}
+        for patch_infos in get_patch_map().values():
+            candidate = getattr(patch_infos[0][0][0], patch_infos[0][0][1])
+            if candidate is not None and isclass(candidate) == estimator:
+                if not estimator or issubclass(candidate, BaseEstimator):
+                    models[patch_infos[0][0][1]] = candidate
+    finally:
+        if patched:
+            unpatch_sklearn()
+
+    return models
+
+
+PATCHED_MODELS = _load_all_models(patched=True)
+UNPATCHED_MODELS = _load_all_models(patched=False)
+
+PATCHED_FUNCTIONS = _load_all_models(patched=True, estimator=False)
+UNPATCHED_FUNCTIONS = _load_all_models(patched=True, estimator=False)
+
+mixin_map = [
+    [
+        ClassifierMixin,
+        ["decision_function", "predict", "predict_proba", "predict_log_proba", "score"],
+        "classification",
+    ],
+    [RegressorMixin, ["predict", "score"], "regression"],
+    [ClusterMixin, ["fit_predict"], "classification"],
+    [TransformerMixin, ["fit_transform", "transform", "score"], "classification"],
+    [OutlierMixin, ["fit_predict", "predict"], "classification"],
+    [KNeighborsMixin, ["kneighbors"], None],
+]
+
+
+SPECIAL_INSTANCES = {
+    i.__str__(): i
```

Comment:
```suggestion
    str(i): i
```

Unless this doesn't work here somehow...